frontend/packages/console-app/src/actions/modify-hpa.ts

```diff
@@ -18,14 +20,24 @@ const hasHPAs = (mapOfResources: RelatedResources) =>
 const hpaRoute = ({ metadata: { name, namespace } }: K8sResourceCommon, kind: K8sKind) =>
   `/workload-hpa/ns/${namespace}/${referenceForModel(kind)}/${name}`;
 
+const isHelmWorkload = (labels: { [key: string]: string }): boolean =>
+  labels?.['app.kubernetes.io/managed-by'] === 'Helm' ||
+  labels?.['heritage'] === 'Helm' ||
+  _.has(labels, 'helm.sh/chart');
+
 export const AddHorizontalPodAutoScaler: KebabAction = (
   kind: K8sKind,
   obj: K8sResourceCommon,
   resources: RelatedResources,
+  customData?: { [key: string]: any },
 ) => ({
   label: `Add ${HorizontalPodAutoscalerModel.label}`,
   href: hpaRoute(obj, kind),
-  hidden: hasHPAs(resources),
+  hidden:
+    hasHPAs(resources) ||
+    isHelmWorkload(obj.metadata.labels) ||
+    resources?.['isOperatorBackedService'] ||
+    isOperatorBackedService(obj, customData?.csvs),
```

Comment:
```suggestion
    isRestrictedWorkload(obj, resources, customData),
```
If you take the previous suggestion we could probably wrap the two other calls `isHelmWorkload` and `isOperatorBackedWorkload`. Would be easier to read and modify if needed.

Function composition is one of the enjoyments of functional development.

Reply:
Yes, I am planning to wrap all the function calls into one. There's one more issue that I came across today, after the PR https://github.com/openshift/console/pull/6236 got merged the HPA actions need to be removed from the context menu as well. I am working on it